services/blockchain-app-registry/shared/utils/downloadRepository.js

```diff
@@ -127,33 +131,30 @@ const getDiff = async (lastSyncedCommitHash, latestCommitHash) => {
 	} catch (error) {
 		let errorMsg = error.message;
 		if (Array.isArray(error)) errorMsg = error.map(e => e.message).join('\n');
-		logger.error(`Unable to get diff due to: ${errorMsg}`);
+		logger.warn(`URL invocation failed: ${url}.`);
+		logger.error(`Unable to get diff due to: ${errorMsg}.`);
 		throw error;
 	}
 };
 
 const filterMetaConfigFilesByNetwork = async (network, filesChanged) => {
-	const filesUpdated = filesChanged.filter(
+	const filesUpdated = (filesChanged || []).filter(
 		file => file.startsWith(network)
 			&& (file.endsWith(FILENAME.APP_JSON) || file.endsWith(FILENAME.NATIVETOKENS_JSON)),
 	);
 	return filesUpdated;
 };
 
 const getUniqueNetworkAppDirPairs = async (files) => {
```

Comment:
As a general practice, please do not make drastic and irrelevant changes to the implementations for an issue intended to write tests. We can create separate issues when necessary.

Same as above.
```suggestion
const getUniqueNetworkAppDirPairs = async (files = []) => {
```

Reply:
This was a small function, hence went with the changes, would have created new issue for a bigger change. Wrote test coverage first and then refactored to ensure same behaviour. Should we create refactoring issue for small functions as well?